pydantic/types.py

```diff
@@ -498,9 +498,6 @@ def __prepare_pydantic_annotations__(
     def __eq__(self, other: Any) -> bool:
         return isinstance(other, self.__class__) and self.get_secret_value() == other.get_secret_value()
 
-    def __hash__(self) -> int:
-        return hash(self.get_secret_value())
```

Comment:
Why is the hash removed?

Reply:
I thought ‍‍‍`__hash__` is defined in `SecretStr` and `SecretBytes` which was not correct.
change reverted and tests added for `SecretStr` and `SecretBytes`